### PR TITLE
Update Db2Adapter.createFhirSchemas to succeed when schemas exist

### PR DESCRIPTION
**Pull request by lmsurpre**
Without this, the step always fails if you've already created a schema
on this db because FHIR_ADMIN already exists. Now it will succeed but
print a WARNING to indicate it already existed.

1. introduced DuplicateNameException and a corresponding helper to
identify when schema creation fails due to the schema already existing

2. renamed DuplicateValueException to UniquenessViolationException to
avoid confusion with DuplicateNameException

3. updated Db2Adapter to catch the DuplicateNameException and WARN
instead of failing

Also added log statement at the end of tenant creation so that you don't
need to scroll to the very top of the output to find the generated key

Signed-off-by: Lee Surprenant <lmsurpre@us.ibm.com>

### CODE REVIEW

**Comment by prb112** [comment]
interesting, the fhir-audit changes are in the PR.

File: fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/api/IDatabaseTranslator.java [hunk 1]
```diff
@@ -50,6 +50,14 @@
      */
     boolean isDuplicate(SQLException x);
     
+    /**
+     * Check the exception to see if it is reporting that THE NAME OF THE OBJECT TO BE CREATED 
+     * OR THE TARGET OF A RENAME STATEMENT IS IDENTICAL TO THE EXISTING NAME OF THE OBJECT TYPE
+     * @param x
+     * @return
+     */
+    boolean isAlreadyExists(SQLException x);
```
Comment (prb112) [inline fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/api/IDatabaseTranslator.java]: the name is a bit convoluted. 

isAlreadyExists  ... may be alternative 
alreadyExists
exists

File: fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/api/UniquenessViolationException.java [hunk 2]
```diff
@@ -9,16 +9,16 @@
 /**
  * Translation of a duplicate key or value constraint SQLException.
  */
-public class DuplicateValueException extends DataAccessException {
+public class UniquenessViolationException extends DataAccessException {
```
Comment (prb112) [inline fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/api/UniquenessViolationException.java]: UniqueConstraintViolationException?

File: fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/api/IDatabaseTranslator.java (hunk 1 above)
Comment (lmsurpre) [reply fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/api/IDatabaseTranslator.java]: yeah, agree its a bit akward.  i went with `is` to match the other methods which are similar in nature.  its saying "is the SQLException an AlreadyExists one".

I had the same feeling about the existing `isDuplicate(SQLException x)` method name as well...

File: fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/api/UniquenessViolationException.java (hunk 2 above)
Comment (lmsurpre) [reply fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/api/UniquenessViolationException.java]: I actual bounced between that and this one.  Will change to that.

File: fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/api/IDatabaseTranslator.java (hunk 1 above)
Comment (prb112) [reply fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/api/IDatabaseTranslator.java]: I have no problemo with it.

**Comment by lmsurpre** [comment]
> interesting, the fhir-audit changes are in the PR.

Sorry, that was just a bit lazy on my part.  I created this PR first, then did those changes on top of these.  Then, instead of cherry-picking it to a new one from master, I just opened that PR against this one.  Just disregard those changes since you already reviewed them (and keep in mind they aren't in master until we merge this one).

File: fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/api/UniquenessViolationException.java (hunk 2 above)
Comment (lmsurpre) [reply fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/api/UniquenessViolationException.java]: done

**Pull request merged**